src/app/meditation/meditation.component.ts

```diff
@@ -98,6 +106,20 @@ export class MeditationComponent {
       this.finishedMeditations = res.filter(data => {
         return data.sittingLeft + data.walkingLeft === 0;
       });
+
+      // Check User meditation status
+      res.forEach(function (current){
```

Comment:
Can you add this to the filter method of `activeMeditations`? That way we reduce another iteration of the array.